X509VerificationUtils.swift

```diff
@@ -0,0 +1,309 @@
+import Foundation
+import ASN1Decoder
+
+enum CertificateValidationStatus: String {
+    case VALID
+    case INVALID_CHAIN
+    case EXPIRED
+    case NOT_YET_VALID
+    case REVOKED
+    case VALIDATION_ERROR
+    case CRL_RETRIEVAL_ERROR
+}
+
+struct CertificateValidationResult {
+    let isValid: Bool
+    let validationStatus: CertificateValidationStatus
+    let errorMessage: String?
+}
+
+class X509VerificationUtils {
+    
+    /// - Parameters:
+    ///   - certChainBase64: Array of Base64 encoded certificates
+    ///   - trustAnchorCertBase64: Base64 encoded trust anchor certificate
+    /// - Returns: CertificateValidationResult with validation status and details
+    static func verifyCertificateChainWithSecTrust(certChainBase64: [String], trustAnchorCertBase64: String) -> CertificateValidationResult {
+        // Pre-check to verify that the last chain certificate and the TA cert are the same
+        if certChainBase64.last != trustAnchorCertBase64 {
```

Comment:
Here we're assuming that the base64 representation is always the same, but it might not be, due to differences in padding, whitespace, or standard vs URL-safe encoding. Would it be more robust to compare the raw bytes instead? What do you think?